MetaMorpheus/EngineLayer/FdrAnalysis/PEPValueAnalysisGeneric.cs

```diff
@@ -87,113 +153,173 @@ public static string ComputePEPValuesForAllPSMsGeneric(List<SpectralMatch> psms,
+
+                                    bool label;
+                                    if (csm.IsDecoy || csm.BetaPeptide.IsDecoy)
+                                    {
+                                        label = false;
+                                        newPsmData = CreateOnePsmDataEntry(searchType, csm, csm.BestMatchingBioPolymersWithSetMods.First().Peptide, 0, label);
+                                    }
+                                    else if (!csm.IsDecoy && !csm.BetaPeptide.IsDecoy && csm.GetFdrInfo(UsePeptideLevelQValueForTraining).QValue <= QValueCutoff)
+                                    {
+                                        label = true;
+                                        newPsmData = CreateOnePsmDataEntry(searchType, csm, csm.BestMatchingBioPolymersWithSetMods.First().Peptide, 0, label);
+                                    }
+                                    else
+                                    {
+                                        continue;
+                                    }
+                                    localPsmDataList.Add(newPsmData);
+                                    localPsmOrder.Add(i);
+                                }
+                                catch (Exception ex)
+                                {
+                                    string message = ex.Message;
+                                }
 
-                    allMetrics.Add(metrics);
-                    sumOfAllAmbiguousPeptidesResolved += ambiguousPeptidesResolved;
-                }
+                                
+                            }
+                            else
+                            {
+                                double bmp = 0;
+                                foreach (var (notch, peptideWithSetMods) in psm.BestMatchingBioPolymersWithSetMods)
+                                {
+                                    bool label;
+                                    double bmpc = psm.BestMatchingBioPolymersWithSetMods.Count();
+                                    if (peptideWithSetMods.Parent.IsDecoy)
+                                    {
+                                        label = false;
+                                        newPsmData = CreateOnePsmDataEntry(searchType, psm,
+                                            peptideWithSetMods, notch, label);
+                                    }
+                                    else if (!peptideWithSetMods.Parent.IsDecoy
+                                        && psm.GetFdrInfo(UsePeptideLevelQValueForTraining).QValue <= QValueCutoff)
+                                    {
+                                        label = true;
+                                        newPsmData = CreateOnePsmDataEntry(searchType, psm,
+                                            peptideWithSetMods, notch, label);
+                                    }
+                                    else
```

Comment:
If I am reading this correctly, we are adding to the training data: from each peptide group, take the best scoring of each full sequence, then add the PSMData if it is a decoy, or if it is a target below q value cutoff. If this is the correct interpretation, is it as intended?

Reply:
Correct interpretation and the intended behavior, yes